app/assets/javascripts/discourse/app/controllers/flag.js

```diff
@@ -18,6 +21,55 @@ export default Controller.extend(ModalFunctionality, {
   topicActionByName: null,
   spammerDetails: null,
 
+  flagActions: {
+    icon: "gavel",
+    label: I18n.t("flagging.take_action"),
+    actions: [
+      {
+        id: "agree_and_keep",
+        icon: "thumbs-up",
+        label: I18n.t("flagging.take_action_options.default.title"),
+        description: I18n.t("flagging.take_action_options.default.details"),
+      },
+      {
+        id: "agree_and_suspend",
+        icon: "ban",
+        label: I18n.t("flagging.take_action_options.suspend.title"),
+        description: I18n.t("flagging.take_action_options.suspend.details"),
+        client_action: "suspend",
+      },
+      {
+        id: "agree_and_silence",
+        icon: "microphone-slash",
+        label: I18n.t("flagging.take_action_options.silence.title"),
+        description: I18n.t("flagging.take_action_options.silence.details"),
+        client_action: "silence",
+      },
+    ],
+  },
+
+  clientSuspend(performAction) {
+    this._penalize("showSuspendModal", performAction);
+  },
+
+  clientSilence(performAction) {
+    this._penalize("showSilenceModal", performAction);
+  },
+
+  async _penalize(adminToolMethod, performAction) {
+    let adminTools = this.adminTools;
```

Comment:
don't think this intermediate variable is very useful

Reply:
The admin tools one? It's also declared like this elsewhere previously - I can remove all instances of that here :+1: